plugins/outputs/yandex_cloud_monitoring/yandex_cloud_monitoring.go

```diff
@@ -93,35 +89,33 @@ func (a *YandexCloudMonitoring) Connect() error {
 	if a.Timeout.Duration <= 0 {
 		a.Timeout.Duration = defaultRequestTimeout
 	}
-
-	a.client = &http.Client{
-		Transport: &http.Transport{
-			Proxy: http.ProxyFromEnvironment,
-		},
-		Timeout: a.Timeout.Duration,
-	}
-
 	if a.EndpointUrl == "" {
 		a.EndpointUrl = defaultEndpointUrl
 	}
+	if a.Service == "" {
+		a.Service = "custom"
+	}
 	if a.MetadataTokenUrl == "" {
 		a.MetadataTokenUrl = defaultMetadataTokenUrl
 	}
 	if a.MetadataFolderUrl == "" {
 		a.MetadataFolderUrl = defaultMetadataFolderUrl
 	}
-	if a.FolderID == "" {
-		folderID, err := getFolderIDFromMetadata(a.client, a.MetadataFolderUrl)
-		if err != nil {
-			return err
-		}
-		a.FolderID = folderID
+
+	a.client = &http.Client{
+		Transport: &http.Transport{
+			Proxy: http.ProxyFromEnvironment,
+		},
+		Timeout: a.Timeout.Duration,
 	}
-	if a.Service == "" {
-		a.Service = "custom"
+
+	var err error
+	a.FolderID, err = a.getFolderIDFromMetadata()
+	if err != nil {
+		return err
 	}
 
-	log.Printf("D! Writing to Yandex.Cloud Monitoring URL: %s", a.EndpointUrl)
+	a.Log.Infof("Writing to Yandex.Cloud Monitoring URL: %s", a.EndpointUrl)
```

Comment:
these should probably be Debugf, as under normal operation the plugins should be rather quiet.

Reply:
I have intentionally made this info-level message -- this is being printed only once during plugin initialization. Seems not so verbose to me. Do you insist?

Reply:
Ok, feel free to leave it.